src/dialogs/quick-open/ha-quick-open-dialog.ts

```diff
@@ -0,0 +1,216 @@
+  }
+
+  private _entityFilterChanged(ev: PolymerChangedEvent<string>) {
+    this._entityFilter = ev.detail.value;
+    if (this._entityFilter.startsWith(">")) {
+      this._commandMode = true;
+      this._entityFilter = this._entityFilter.substring(1);
+    } else {
+      this._commandMode = false;
+    }
+
+    let items: QuickOpenItem[];
+    if (this._commandMode) {
+      items = this._generateCommandItems(this._entityFilter);
+    } else {
+      items = this._generateEntityItems(this.hass, this._entityFilter);
+    }
+    this._items = this._filterItems(items, this._entityFilter);
+  }
+
+  private _generateCommandItems(_filter: string): QuickOpenItem[] {
+    const reloadableDomains = componentsWithService(this.hass, "reload").sort();
+
+    return reloadableDomains.map((domain) => ({
+      text:
+        this.hass.localize(`ui.dialogs.quick_open.commands.${domain}`) ||
+        this.hass.localize(
+          "ui.dialogs.quick_open.commands.reload",
+          "domain",
+          domainToName(this.hass.localize, domain)
+        ),
+      domain,
+      onClick: this._executeCommand,
+    }));
+  }
+
+  private _generateEntityItems(
+    hass: HomeAssistant,
+    _filter: string
+  ): QuickOpenItem[] {
+    return Object.keys(hass.states)
+      .map((key) => hass.states[key])
+      .map<QuickOpenItem>(({ entity_id }) => ({
+        text: entity_id,
+        domain: computeDomain(entity_id),
+        onClick: this._entityMoreInfo,
+      }));
+  }
+
+  private _filterItems(
+    items: QuickOpenItem[],
+    filter: string
+  ): QuickOpenItem[] {
+    return items
+      .filter(({ text }) => fuzzySequentialMatch(filter.toLowerCase(), text))
+      .sort((itemA, itemB) => {
+        if (itemA.text < itemB.text) {
+          return -1;
+        }
+        if (itemA.text > itemB.text) {
+          return 1;
+        }
+        return 0;
+      });
+  }
+
+  private async _executeCommand(ev: Event) {
+    const eventData = {
+      domain: (ev.target as any).domain,
+      service: "reload",
+    };
+
+    await this.hass.callService(eventData.domain, "reload");
+
+    fireEvent(this, "hass-service-called", eventData);
+
+    this.closeDialog();
+  }
+
+  private _entityMoreInfo(ev: Event) {
+    ev.preventDefault();
+    fireEvent(this, "hass-more-info", {
+      entityId: (ev.target as any).text,
+    });
+    this.closeDialog();
+  }
+
+  protected render() {
+    if (!this._opened) {
+      return html``;
+    }
+
+    return html`
+      <ha-dialog open @closed=${this.closeDialog} hideActions>
+        <paper-input
+          dialogInitialFocus
+          no-label-float
+          @value-changed=${this._entityFilterChanged}
+          .label=${this.hass.localize(
+            "ui.dialogs.quick_open.filter_placeholder"
+          )}
+          type="search"
+          value=${this._commandMode
+            ? `>${this._entityFilter}`
+            : this._entityFilter}
+        ></paper-input>
```

Comment:
Make this the header of the dialog, so it doesn't scroll with the content.

Use the `heading` slot <https://github.com/home-assistant/frontend/blob/dev/src/dialogs/more-info/ha-more-info-dialog.ts#L109-L113>

Reply:
After doing that, it looks like there might now be too much space between the text input and the first result. The gap comes from the default dialog-content-padding of 20px.

Any suggestions? I could change dialog-content-padding to compensate, or maybe override it with a wrapping div?

![image](https://user-images.githubusercontent.com/1480827/95237175-bb25e100-07bc-11eb-894c-0c4d60c6023f.png)

Reply:
Pushed the changes, but leaving this comment unresolved until we have a solution for the extra whitespace.

Reply:
You can remove (or make it smaller) the top padding of the dialog content.

Reply:
Fixed:
<img src="https://user-images.githubusercontent.com/1480827/95283852-927c0680-0811-11eb-8982-750c17207a22.png" width="50%" />